sound basics lesson content (#39)
ecoacoustics/website: content/resources/lessons/sound-basics/practical/index.md

---
title: Practical
weight: 3
---

This is a simple exercise to help you become familiar with changing the window
size of a sound file using Audacity. This will help you to understand how
changing the window size can alter the frequency and temporal resolution of your
spectrograms, and affect how your spectrograms will look visually.

1. Find `20220421T100000+1000_SEQP-Samford-Wet-B_644370.flac` in the 2022
   Ecoacoustics Symposium/SERF Data folder.
2. Open Audacity, and drag the sound file into Audacity to open it.
![](20221206111634.png)

> 1. We can view the spectrogram of our sound file by accessing the drop down arrow menu, and

Reviewer: Which drop down menu? Maybe:

> At the top-left of each track, there is a drop-down menu. It's represented by a ▼ icon. Click on it, and then click on _Spectrogram_ to switch the track view to the spectrogram mode.

   clicking _Spectrogram_.
![](20221206111806.png)
1. You should now see the spectrogram view. Going back to the same drop down menu, we can

>    now select _Spectrogram Settings_, which brings up this window:

Reviewer: be specific:
```suggestion
   now select _Spectrogram Settings_, which brings up the _spectrogram settings_ window:
```

   ![](20221206112035.png)  
2. Take note of some important information that we should be familiar with by now:

>    1. Max Frequency: 11025 Hz. Remember that Max Frequency is half of the sampling frequency. Therefore,

Reviewer: not correct, this setting is basically just a y-max on the graph. It is not related to the Nyquist at all - though we should advise users to set this to their Nyquist by default

Reviewer: playing with this and min is how we recreate the y component of raven's so-called "zoom"

>    1. Max Frequency: 11025 Hz. Remember that Max Frequency is half of the sampling frequency. Therefore,

Reviewer: > half of the sampling frequency

Use the proper term, _Nyquist_ - expand on it for clarity if needed

      this file was recorded with a sampling frequency of 22050 Hz. 

>    2. Window size: 2048

Reviewer: missing context?

The default settings on a default install of audacity may not have this set to 2048 (it might, i haven't checked). The language here should represent that the window size could be any value.

Might be a good time to recap window tradeoffs.

3. Lets try to figure out what the temporal and frequency resolution is for this

>    file, with these default settings. Refer back to the Presentation if you are

Reviewer: link back to the presentation, and no capital _P_ for presentation

   unsure how to work these out.
   1. TR = Window size / SR

>    2. FR = SR / window size

Reviewer: If this is meant to be math, use MathJax

>    2. FR = SR / window size

Reviewer: don't use abbreviations, use words

4. Now, lets change the window size to 256. Click on the _Window size_ input
   box, and select the new value.
   ![](20221206113827.png)  
5. Calculate the new temporal and frequency resolution values - did they

>    increase or decrease?

Reviewer: use a `<details>` to show/ihide the answers https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details

6. What about the spectrogram, does it look different?

> Feel free to keep playing around with the values, and understanding the effects.

Reviewer: ```suggestion
Keep playing around with the spectrogram settings. Try to notice the effects each setting has.
```

Personal style: polite/passive/deferring
Professional: Lot of passiveness/politeness is noise. Avoid it. Issue instructions.

---
title: Practical
weight: 3
---

This is a simple exercise to help you become familiar with changing the window
size of a sound file using Audacity. This will help you to understand how
changing the window size can alter the frequency and temporal resolution of your
spectrograms, and affect how your spectrograms will look visually.

> 

Reviewer: Add a setup section, e.g.:

https://github.com/ecoacoustics/website/blob/cfafde4899d40ac4e2fc5f276d77f00deb528d73/content/resources/lessons/wrangling-sound-files/practical/index.md?plain=1#L15-L17

3. At the top-left of each track, there is a drop-down menu. It's represented by
   a ▼ icon. Click on it, and then click on _Spectrogram_ to switch the track
   view to the spectrogram mode.
   ![Drop down menu view of sound file](20221206111806.png)
4. You should now see the spectrogram view. Going back to the same drop down
   menu, we can now select _Spectrogram Settings_, which brings up the
   _spectrogram settings_ window:
   ![Spectrogram settings view](20221206112035.png)
5. Inside _Spectrogram Settings_, you can adjust the scale, colour, and
   algorithm settings for the spectrogram view. Take note of the scale settings,
   Min Frequency and Max Frequency (Hz). Adjusting these values will change the
   y axis of the spectrogram view, and is similar to the "vertical zoom" in
   Raven.
    1. We recommend starting with a Max Frequency that is equal to the Nyquist
       frequency of your recordings. Remember that the Nyquist Frequency is half
       of the sampling frequency. This file was recorded with a sampling
       frequency of 22050 Hz, and therefore has a Nyquist Frequency of 11025 Hz.
6. Moving on to the algorithm settings, take particular note of the window size.

>    This is the number of samples that are used to calculate the spectrogram.

Reviewer: ```suggestion
   This is the number of samples that are used to calculate each frame of the spectrogram.
```

   Recall from the presentation that temporal and frequency resolution of the
   spectrogram is related to the window size, which results in a trade-off

>    between frequency versus temporal resolution. The larger the window size, the

Reviewer: ```suggestion
   between frequency and temporal resolution. The larger the window size, the
```

   more samples are used to calculate the spectrogram, and the more frequency
   resolution you will have. However, the smaller the window size, the more
   temporal resolution you will have.

> 7. Question: Set the window length value to 2048. What is the temporal

Reviewer: ```suggestion
7. Question: Set the window length value to `2048`. What is the temporal
```

   and frequency resolution for this file at this setting? See the following
   formulas for help, or refer back to the [presentation]({{% relref
   "presentation" %}}) for more information.

{{% katex display %}} Temporal\:resolution = \frac{Window\:size}{Sampling\:rate} {{% /katex %}}
{{% katex display %}} Frequency\:resolution = \frac{Sampling\:rate}{Window\:size} {{% /katex %}}

<details>
    <summary>Answer:</summary>
Temporal resolution = 0.093 seconds, Frequency resolution = 10.767 Hz.
</details>

> 8. Now, change the window size to 256. Click on the _Window size_ input

Reviewer: ```suggestion
8. Now, change the window size to `256`. Click on the _Window size_ input
```
